cadc-storage-permissions/src/intTest/java/org/opencadc/inventory/permissions/PermissionsClientTest.java

```diff
@@ -0,0 +1,196 @@
+ *  OpenCADC project.                    OpenCADC.
+ *
+ *  OpenCADC is free software:           OpenCADC est un logiciel libre ;
+ *  you can redistribute it and/or       vous pouvez le redistribuer ou le
+ *  modify it under the terms of         modifier suivant les termes de
+ *  the GNU Affero General Public        la “GNU Affero General Public
+ *  License as published by the          License” telle que publiée
+ *  Free Software Foundation,            par la Free Software Foundation
+ *  either version 3 of the              : soit la version 3 de cette
+ *  License, or (at your option)         licence, soit (à votre gré)
+ *  any later version.                   toute version ultérieure.
+ *
+ *  OpenCADC is distributed in the       OpenCADC est distribué
+ *  hope that it will be useful,         dans l’espoir qu’il vous
+ *  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+ *  without even the implied             GARANTIE : sans même la garantie
+ *  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+ *  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+ *  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+ *  General Public License for           Générale Publique GNU Affero
+ *  more details.                        pour plus de détails.
+ *
+ *  You should have received             Vous devriez avoir reçu une
+ *  a copy of the GNU Affero             copie de la Licence Générale
+ *  General Public License along         Publique GNU Affero avec
+ *  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+ *  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+ *                                       <http://www.gnu.org/licenses/>.
+ *
+ *  : 5 $
+ *
+ ************************************************************************
+ */
+
+package org.opencadc.inventory.permissions;
+
+import ca.nrc.cadc.auth.SSLUtil;
+import ca.nrc.cadc.util.FileUtil;
+import ca.nrc.cadc.util.Log4jInit;
+import java.net.URI;
+import java.security.PrivilegedExceptionAction;
+import java.util.List;
+import javax.security.auth.Subject;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.opencadc.gms.GroupURI;
+
+public class PermissionsClientTest {
+
+    private static final Logger log = Logger.getLogger(PermissionsClientTest.class);
+
+    private static Subject cadcAnonTest1Subject;
+    private static Subject cadcRegTest1Subject;
+    private static URI serviceID;
+
+    private static URI testArtifact;
+    private static GroupURI readGroup1;
+    private static GroupURI writeGroup1;
+    private static GroupURI writeGroup2;
+    private static GroupURI writeGroup3;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception
+    {
+        Log4jInit.setLevel("org.opencadc.inventory.permissions", Level.INFO);
+
```

Comment:
I don't know who made up cadcanontest1 but if there is a certificate it is not anonymous.

1. Don't do a Subject.doAs at all for anon
2. Use cadcreegtest1 for auth but permission denied
3. Use cadcauthtest1 for auth and allowed

Reply:
Updated.